src/Internal/Service/Blockfrost.purs

```diff
@@ -253,11 +379,360 @@ getTxMetadata txHash = do
       | otherwise -> Right metadata
 
 --------------------------------------------------------------------------------
--- `getTxMetadata` reponse parsing
+-- Get utxos at address / by output reference
+--------------------------------------------------------------------------------
+
+utxosAt :: Address -> BlockfrostServiceM (Either ClientError UtxoMap)
+utxosAt address = runExceptT $
+  ExceptT (utxosAtAddressOnPage 1)
+    >>= (ExceptT <<< resolveBlockfrostUtxosAtAddress)
+  where
+  utxosAtAddressOnPage
+    :: Int -> BlockfrostServiceM (Either ClientError BlockfrostUtxosAtAddress)
+  utxosAtAddressOnPage page = runExceptT do
+    -- Maximum number of results per page supported by Blockfrost:
+    let maxNumResultsOnPage = 100
+    utxos <- ExceptT $ handleBlockfrostResponse <$>
+      blockfrostGetRequest (UtxosAtAddress address page maxNumResultsOnPage)
+    case Array.length (unwrap utxos) < maxNumResultsOnPage of
+      true -> pure utxos
+      false -> append utxos <$> ExceptT (utxosAtAddressOnPage $ page + 1)
```

Comment:
Yes, but it's impossible to write an `ExceptT` type signature with a partially applied type synonym like `BlockfrostServiceM`, and expanding it would be even uglier IMO.

Reply:
Right, forgot that it was a synonym here :(